packages/components/src/aspect-ratio/component.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { css } from '@emotion/react';
+
+/**
+ * WordPress dependencies
+ */
+import { cloneElement, Children, isValidElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	PolymorphicComponentProps,
+} from '../ui/context';
+import type { AspectRatioProps } from './types';
+import * as styles from './styles';
+import { useCx } from '../utils/hooks';
+
+const { AspectRatioResizer, AspectRatioView } = styles;
+
+function AspectRatio(
+	props: PolymorphicComponentProps< AspectRatioProps, 'div' >,
+	forwardedRef: import('react').Ref< any >
+) {
+	const {
+		children,
+		className,
+		ratio = 1,
+		width,
+		...otherProps
+	} = useContextSystem( props, 'AspectRatio' );
+	const cx = useCx();
+
+	const [ clonedChild ] = Children.map( children, ( child ) => {
+		if ( ! isValidElement( child ) ) {
+			return child;
+		}
+		return cloneElement( child, {
+			...child.props,
+			className: cx( styles.content, child.props.className ),
+		} );
+	} );
```

Comment:
If I understand correctly:
1. We're cloning the `children` so that we can filter out non-React elements
2. Only the first element of the resulting array (`clonedChild`) ends up being added to the DOM.

If this is the case, I believe that we should at least add some inline comments and probably an explanation in the README about the fact that only the first valid `ReactElement` will be actually rendered

Reply:
Actually the implementation was a bit worse. It would return the first element even if it was invalid. I updated here: https://github.com/WordPress/gutenberg/pull/33603/commits/f0f87eca41e0f6cc3b7133fad44e75ec5a711d67. But need some help with the type compatibility there..